src/source/mod.rs

```diff
@@ -442,7 +443,7 @@ pub trait Source: Iterator<Item = Sample> {
     /// let source = source.buffered().reverb(Duration::from_millis(100), 0.7);
     /// ```
     #[inline]
-    fn reverb(self, duration: Duration, amplitude: f32) -> Mix<Self, Delay<Amplify<Self>>>
+    fn reverb(self, duration: Duration, amplitude: AmpFactor) -> Mix<Self, Delay<Amplify<Self>>>
```

Comment:
I see why you want for an `enum`. We could still add a method `reverb_decibel`.

Alternatively maybe a builder like approach could be nice. Reverb gets to be its own type and this function no longer takes an amplitude argument. Instead we use a default value and you set the amplitude later with a setter. There are then two setters mirroring, a log/decibel one and a linear one.